src/ExecuteTimeWidget.ts

```diff
@@ -346,6 +351,7 @@ export default class ExecuteTimeWidget extends Widget {
       .composite as boolean;
     this._settings.historyCount = settings.get('historyCount')
       .composite as number;
+    this._settings.dateFormat = settings.get('dateFormat').composite as string;
```

Comment:
The format string should be validated and a useful message shown to the user. Currently, it will error out later silently, but no indication of the error will be shown except in the developer console:

```js
RangeError: Format string contains an unescaped latin alphabet character `r`
    at index.js:390:1
    at Array.map (<anonymous>)
    at format (index.js:370:4)
    at getTimeString (formatters.js:6:1)
    at ExecuteTimeWidget._updateCodeCell (ExecuteTimeWidget.js:213:47)
```

I would suggest creating a new function called `isValidFormat` that would attempt to format a hard-coded date in try-catch and returned the error if caught. Then we could display the error message using `showDialog` from `@jupyterlab/apputils`.

Alternatively, one could try to define a regex in schema to perform validation directly the in the form.

Reply:
That is a good suggestion.

validating using regex would be a quick and simple solution since it will be looking at the characters in the date field unicode tokens.

I will look into the displaying error message using `showDialog` since I am unfamiliar with it.

Reply:
@krassowski which value did you use that got you that error?

Reply:
> @krassowski which value did you use that got you that error
I have found a good test case. Apologies